src/fn/move-contacts.js

```diff
@@ -21,6 +22,29 @@ module.exports = {
   }
 };
 
+const moveReportsBatch = async (db, contactIds, skip, replacementLineage, contactId, options) => {
+  const queryOptions = {
+    keys: contactIds.map(id => [`contact:${id}`]),
+    include_docs: true,
+    limit: BATCH_SIZE,
+    skip: skip,
+  };
+  const result = await db.query('medic-client/reports_by_freetext', queryOptions);
+  const reportDocs = result.rows.map(row => row.doc);
+  info(`Processing ${skip} to (${skip + BATCH_SIZE}) docs of ${result.total_rows} total`);
+  const updatedReports = replaceLineageInReports(reportDocs, replacementLineage, contactId);
+  updatedReports.forEach(updatedDoc => {
+    lineageManipulation.minifyLineagesInDoc(updatedDoc);
+    writeDocumentToDisk(options, updatedDoc);
+  });
+
+  if(result.total_rows < skip + BATCH_SIZE) {
+    return result.total_rows;
+  }
```

Comment:
`total_rows` returns the total rows in the view, not just how many rows apply to your filter. 
For this to be correct, it should be changed to:
```suggestion
  if (result.rows.length < BATCH_SIZE) {
    return skip + result.rows.length;
  }
```